src/framework/components/anim/component.js

```diff
@@ -0,0 +1,441 @@
+Object.assign(pc, function () {
+
+    /**
+     * @class
+     * @name pc.AnimComponentLayer
+     * @classdesc The Anim Component Layer allows managers a single layer of the animation state graph.
+     * @description Create a new AnimComponentLayer.
+     * @param {string} name - The name of the layer.
+     * @param {object} controller - The controller to manage this layers animations.
+     * @param {pc.AnimComponent} component - The component that this layer is a member of.
+     */
+    var AnimComponentLayer = function (name, controller, component) {
+        this._name = name;
+        this._controller = controller;
+        this._component = component;
+    };
+
+    Object.assign(AnimComponentLayer.prototype, {
+        /**
+         * @function
+         * @name pc.AnimComponentLayer#play
+         * @description Start playing the animation in the current state.
+         * @param {string} name - If provided, will begin playing from the start of the state with this name.

```

Comment:
Would be nice to mark `name` as optional so that the proper typings are generated.
```
* @param {string} [name] - If provided, will begin playing from the start of the state with this name.
```

Reply:
Updated this, cheers 👍